sklearn/cluster/tests/test_hierarchical.py

```diff
@@ -598,3 +598,15 @@ def increment(self, *args, **kwargs):
     linkage_tree(X, connectivity=connectivity, affinity=fa.increment)
 
     assert_equal(fa.counter, 3)
+
+
+def test_n_components_deprecation():
+    # Test that a Deprecation warning is thrown when n_components_
+    # attribute is called from AgglomerativeClustering.fit() method
```

Comment:
Not testing anything about the fit method. Instead say "when n_components_ attribute is accessed"

Reply:
Thanks, should be solved by commit #260a15b